### ISSUE TITLE
BUG: Polyfit in IVIM TRR returns Negative Values

### ISSUE THREAD

**Issue by ShreyasFadnavis** [dipy/dipy#2050]
Following up with this issue with a PR!

Potential fixes:
- Use `scipy.optimize.lsq_linear` and set the bounds manually
or 
- Use `scipy.optimize.nnls`

I assume `nnls` will be faster as it is running FORTAN under the hood.

Any other suggestions?

**Comment by ShreyasFadnavis** [comment]
Any other suggestions anyone? @arokem @skoudoro @RafaelNH ?

**Comment by skoudoro** [comment]
Good catch and +1 for `scipy.optimize.nnls`

**Comment by arokem** [comment]
Yes. NNLS should be the way to go. I'm not sure about performance, but it will probably be more correct than trying to implement the bounds by ourselves.
